src/libsyntax/ext/tt/macro_rules.rs

```diff
@@ -364,7 +364,24 @@ pub fn compile(sess: &ParseSess, features: &Features, def: &ast::Item, edition:
     });
 
     if body.legacy {
-        let allow_internal_unstable = attr::contains_name(&def.attrs, "allow_internal_unstable");
+        let allow_internal_unstable = attr::find_by_name(&def.attrs, "allow_internal_unstable")
+            .map_or(Vec::new(), |attr| attr
+                .meta_item_list()
+                .map(|list| list.iter()
+                    .map(|it| it.name().unwrap_or_else(|| sess.span_diagnostic.span_bug(
+                        it.span, "allow internal unstable expects feature names",
+                    )))
+                    .collect()
+                )
+                .unwrap_or_else(|| {
+                    sess.span_diagnostic.span_warn(
+                        attr.span, "allow_internal_unstable expects list of feature names. In the \
+                        future this will become a hard error. Please use `allow_internal_unstable(\
+                        foo, bar)` to only allow the `foo` and `bar` features",
+                    );
```

Comment:
This warning is unnecessary, it's already reported by the generic attribute validation machinery.
You'll have to remove `Word` from the `template!(..)` for `allow_internal_unstable` though and add it to the `should_warn` list in `fn check_builtin_attribute`.

Reply:
Oh this is awesome!

Reply:
Or not. It doesn't do the same thing. Because it's a lint and the compiler's default setting of `-Dwarnings` crates like stdsimd fail to compile. It would be a big pain to transition everything at once.

Reply:
Ugh.
Ok, then we just need to make it an error once stdsimd/etc is fixed.